_docs/tasks/traffic-management/circuit-breaking.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Circuit Breaking
+overview: This task demonstrates the circuit-breaking capability for resilient applications
```

Comment:
one thought: Envoy's notion of circuit breaker is pretty confusing for folks from the netflix land. In fact, envoy's circuit breaker is close to bulk head/connection pooling, while the outlier detection is close to the circuit breaker that people have become familiar with.

In the new v1alpha2 APIs, I have renamed the circuit breaker to ConnectionPools and the outlier detection is called outlier detection. They are two separate things. Do you think it makes sense to rename connection pool to bullhead, and outlier detection to circuit breaker? And if we do so, this doc would be bulk heads and circuit breakers. Though in strict sense, its not really a bulk head, because Envoy's connection pool is per upstream host in the cluster (not per cluster). Also, the concept of a bulk head in an event driven sidecar does not really click, as bulk heads are typically useful when you do synchronous calls, with one thread per request. When you have a shared thread pool serving an  event stream, its something else.

Let me know what you think. I am okay just calling this circuit breaker if it simplifies the message for the audience.

Reply:
These are great points. One thing I did like about the current config of "circuit breaker" is that it kinda abstracts/hides away the impedance mismatch between hystrix and envoy style circuit breaking -- though the abstraction is a bit leaky. I do like the idea to make "Bulkhead" and "Circuit breaking" more distinct in the configs vs connection throttling/pooling/outlier detection, etc. Guess for reactive calls, a bulkhead would just be a semaphore to count # of outstanding requests, etc. 

What if we have something like "circuit breaker" which is really outlier detection and then "connection limits" which is really envoy's circuit breaker?

Reply:
Thats basically what I have in v1alpha2. ConnectionPool (which is circuit breaker), and Outlier Detection (which is outlier detection). I could rename Outlier Detection to Circuit Breaker if it helps the messaging. Let me know and I will fix the istio/api

Reply:
Yah @rshriram i think Outlier Detection to Circuit breaker would be a good move.